### PR TITLE
Edit & delete log functionality

### LINKED ISSUE #85: [FE] Edit + delete logs
### Description
Users want to be able to edit and delete the logs they create.

**Unfinished parts of this ticket exist on branch** `jai-85-edit-delete-logs`. You may want to use this as reference, but you can also start from scratch. 

**Edit**
- When a user clicks the Edit button on a Log, populate the Log Modal with the contents of the clicked log
- On Save, send a `PATCH` to `/api/logs/:logId` to update the log
- Make a react-hot-toast based on the result of the request
- Make sure the Edit button only appears on log's that the user has authored

**Delete**
- When a user clicks the Delete button, display a confirmation modal
- if the user clicks Confirm, make a request to `DELETE /api/logs/:logId`
- Make a react-hot-toast based on the result of the request
- Make sure the Delete button only appears on logs that the user has authored

### Acceptance Criteria
- [ ]  Proper error handling
- [ ]  Implements Figma design
- [ ]  Proper JS doc comments above functions

### Other Notes
- N/a

### PR DESCRIPTION
**Pull request by ShmulTomer**
## PR Title

Issue Number(s): #85
What does this PR change and why?

Thoughts:
- sshould test if edit/delete works b/c same user email, but logged in Google vs. non-Google sign in (can't test since Google sign in isnt working for me anymore atm)
- After creating a new log with tags, "recent dog tags" don't show on dashboard

- Added cancel confirmation for both creating and editing a log

Delete Functionality:
- delete log modal
- shows toast on deletion and removes log from screen
- only the author can delete

Edit Functionality:
- uses the LogModal
- shows toast on edit
- only the author can edit

### Checklist

- [x] Requirements have been implemented
- [x] Acceptance criteria is met
- [x] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-a841a38d02db4428a7cdc85dcbb8a35c?pvs=4) have been updated or are not necessary
- [x] Relevant reviewers (EM) have been assigned to this PR

### Critical Changes

- Database change / migration to run
- Environment config change
- Breaking API change

### Related PRs

PR Number(s): #number_of_pr

### How To Test

The steps to test the functionality of your ticket. Include edge cases and testing of error handling, if applicable.

### CODE REVIEW

File: src/components/Log/LogModal.jsx [hunk 1]
```diff
@@ -30,8 +30,69 @@ export default function LogModal({ dogId, userId, onClose, onSubmit }) {
...
+    }, {});
+  };
+  
+
+  // Updates the logData state with the log data from the database when editing
   useEffect(() => {
+
+    if (logId && !retrievedLog) {
+      
+      fetch("/api/logs/" + logId)
```
Comment (afazio1) [inline src/components/Log/LogModal.jsx]: if you pass in the entire `log` object instead of just `logId` as props to this component, you won't have to fetch the log again. Sometimes it takes too long for the log contents to populate in the modal, which leaves the fields blank and is confusing for the user. If you would rather rely on the server state, I would disable the inputs or indicate a loading state in some way (we can talk to design about this).

**Review by afazio1 — CHANGES_REQUESTED** [review]
Awesome work! Thanks for getting this out so quick. 

Some things I found:
- fetching log info for the edit modal sometimes takes too long, which leaves the modal blank and can be confusing for the user. (see comment in code)
- when editing a log, the first time you click on any dropdown, the dropdown will not be populated with state of the currently selected values
- I found an issue on the deploy preview (but not locally for some reason) where a toast kept appearing when clicking a specific button on one of the modals, but I changed the button to `type=button` and it resolved the issue

File: src/components/Tab/TabContainer.jsx [hunk 2]
```diff
@@ -34,9 +35,22 @@ export default function TabContainer({
...
   filteredLogs,
+  onEditLog,
+  onDeleteLog
 }) {
   const router = useRouter();
 
+  const { data: session } = useSession();
+  const user = session?.user;
+
+  const handleEditSuccess = (success) => {
```
Comment (afazio1) [inline src/components/Tab/TabContainer.jsx]: do we need this anymore? seems like the state updates properly

File: src/components/Log/Log.jsx [hunk 3]
```diff
@@ -13,8 +23,107 @@ export default function Log({ log }) {
...
+            }}
+              
+            onClose={() => {
+              setShowDeleteModal(false);
+            }}>
+
+          </DeleteLogModal>
+          :
+          <></>}
+      {isAuthor &&(
```
Comment (afazio1) [inline src/components/Log/Log.jsx:116]: I would check the spacing on these icons and refer to the figma

**Comment by ShmulTomer** [comment]
Changes:
- if the author of a log no longer exists in “user”, it gave an error -> now it just shows N/A for the author
- edit and delete spacing better matches the Figma
- passed in log so editing a log populates instantly
(remnant from the unfinished branch, thanks for pointing this out)
- populating the dropdown menu when clicked the first time (fixed by editing DropdownMenu itself, updating the enabled options whenever selectedOptions prop is changed)

**Review by afazio1 — APPROVED** [review]
Great work!

**Comment by laurenmcp** [comment]
Looks great to me. Thanks for the good work Tomer!!

**Review by laurenmcp — APPROVED** [review]

**Comment by dograh** [comment]
The edit and delete buttons look great! I just have a few minor changes. Can we remove the word "a" from the edit a log title and keep it as "Edit Log"? Can we also increase the spacing between the text and cancel/confirm buttons in the delete confirmation modal? Other than that it looks perfect. Thanks for your hard work!

**Comment by ShmulTomer** [comment]
Made those changes! Thanks for reviewing

@afazio1 do you merge or do I?

**Comment by afazio1** [comment]
I will merge👍

**Review by afazio1 — APPROVED** [review]
